sktime/split/expandinggreedy.py

```diff
@@ -59,18 +64,29 @@ def __init__(self, test_size: int, folds: int = 5, step_length: int = None):
         self.step_length = step_length
         self.fh = np.arange(test_size) + 1
 
+        # no algorithm implemented that is faster for float than naive iteration
+        if isinstance(test_size, float):
+            self.set_tags(**{"split_hierarchical": False})
+
     def _split(self, y: pd.Index) -> SPLIT_GENERATOR_TYPE:
+        test_size = self.test_size
+
+        if isinstance(test_size, float):
+            _test_size = round(len(y) * test_size)
```

Comment:
I saw you're using floor in #5330, so just a suggestion: `math.floor` instead of `round`?

Reply:
for test set, I think it should be `ceil` for consistency, but agreed

Reply:
fixed